mleap-tensor/src/test/scala/ml/combust/mleap/tensor/TensorSpec.scala

```diff
@@ -80,16 +167,65 @@ class TensorSpec extends FunSpec {
       assert(tensor1 != tensor2)
     }
 
-    it("should return true for sparse tensors with dimension -1 and same elements") {
-      val tensor1 = Tensor.create(Array(22.3, 45.6, 99.3), Seq(-1), Some(Seq(Seq(1, 2, 4))))
-      val tensor2 = Tensor.create(Array(22.3, 45.6, 99.3), Seq(-1), Some(Seq(Seq(1, 2, 4))))
+    it("should raise exception for sparse tensors with dimension -1") {
+      assertThrows[IllegalArgumentException] {
+        Tensor.create(Array(22.3, 45.6, 99.3), Seq(-1), Some(Seq(Seq(1, 2, 4))))
+      }
+    }
+
+  }
+
+  describe("SparseTensor") {
+    val random = new scala.util.Random
+    val indices = Seq(Seq(0, 0, 0), Seq(1, 1, 0), Seq(2, 1, 0))
+    val array = Array.fill[Float](indices.length) {
+      random.nextFloat()
+    }
+    val shape = Seq(5, 4, 3)
+    val otherShape = Seq(3, 2, 1)
+
+    it("should return true for empty sparse tensors") {
+      val tensor1 = Tensor.create(Array(), shape, Some(Seq(Seq())))
+      val tensor2 = Tensor.create(Array(), shape, Some(Seq(Seq())))
+      assert(tensor1 == tensor2)
+    }
+
+    it("should return true for sparse tensors with same elements and dimensions") {
+      val tensor1 = Tensor.create(array, shape, Some(indices))
+      val tensor2 = Tensor.create(array, shape, Some(indices))
       assert(tensor1 == tensor2)
     }
 
-    it("should return false for sparse tensors with dimension -1 but different elements") {
-      val tensor1 = Tensor.create(Array(22.3, 45.6, 99.3), Seq(-1), Some(Seq(Seq(1, 2, 4))))
-      val tensor2 = Tensor.create(Array(22.3, 95.6, 99.3), Seq(-1), Some(Seq(Seq(1, 2, 4))))
+    it("should return false for sparse tensors with different dimensions") {
+      val tensor1 = Tensor.create(array, shape, Some(indices))
+      val tensor2 = Tensor.create(array, otherShape, Some(indices))
+      assert(tensor1 != tensor2)
+    }
+
+    it("should return false for sparse tensors with same dimension but different elements") {
+      val tensor1 = Tensor.create(array, shape, Some(indices))
+      val tensor2 = Tensor.create(array.map(x => x * 2), shape, Some(indices))
       assert(tensor1 != tensor2)
     }
+
+    it("should raise exception for sparse tensors with dimension -1") {
```

Comment:
I think this duplicates a test that we have above

Reply:
one is vector, the other one is Tensor
Because the indices is different